From 0e4906f876a1a3bb8b70591dde9845e833f443b7 Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Mon, 2 Jun 2014 18:11:44 -0400
Subject: [PATCH] libarchive: add host-libarchive

This pulls in the libarchive updates from upstream buildroot. It
includes host-libarchive and a version bump.
---
 package/libarchive/Config.in       |  6 ++++++
 package/libarchive/libarchive.hash |  2 +-
 package/libarchive/libarchive.mk   | 31 ++++++++++++++++++++++++++++++-
 3 files changed, 37 insertions(+), 2 deletions(-)

diff --git a/package/libarchive/Config.in b/package/libarchive/Config.in
index 81b5936..fd429ac 100644
--- a/package/libarchive/Config.in
+++ b/package/libarchive/Config.in
@@ -24,4 +24,10 @@ config BR2_PACKAGE_LIBARCHIVE_BSDCPIO
 	  The 'bsdcpio' program is a different interface to
 	  essentially the same functionality as 'bsdtar'.
 
+config BR2_PACKAGE_LIBARCHIVE_BSDCAT
+	bool "bsdcat"
+	help
+	  The 'bsdcat' program is a simple replacement tool
+	  for zcat, bzcat, xzcat, and such.
+
 endif
diff --git a/package/libarchive/libarchive.hash b/package/libarchive/libarchive.hash
index 5f96211..2c39bd3 100644
--- a/package/libarchive/libarchive.hash
+++ b/package/libarchive/libarchive.hash
@@ -1,2 +1,2 @@
 # Locally computed:
-sha256  eb87eacd8fe49e8d90c8fdc189813023ccc319c5e752b01fb6ad0cc7b2c53d5e  libarchive-3.1.2.tar.gz
+sha256  72ee1a4e3fd534525f13a0ba1aa7b05b203d186e0c6072a8a4738649d0b3cfd2  libarchive-3.2.1.tar.gz
diff --git a/package/libarchive/libarchive.mk b/package/libarchive/libarchive.mk
index 726142b..366f1ea 100644
--- a/package/libarchive/libarchive.mk
+++ b/package/libarchive/libarchive.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-LIBARCHIVE_VERSION = 3.1.2
+LIBARCHIVE_VERSION = 3.2.1
 LIBARCHIVE_SITE = http://www.libarchive.org/downloads
 LIBARCHIVE_INSTALL_STAGING = YES
 LIBARCHIVE_LICENSE = BSD-2c, BSD-3c
@@ -30,6 +30,16 @@ else
 LIBARCHIVE_CONF_OPTS += --disable-bsdcpio
 endif
 
+ifeq ($(BR2_PACKAGE_LIBARCHIVE_BSDCAT),y)
+ifeq ($(BR2_STATIC_LIBS),y)
+LIBARCHIVE_CONF_OPTS += --enable-bsdcat=static
+else
+LIBARCHIVE_CONF_OPTS += --enable-bsdcat=shared
+endif
+else
+LIBARCHIVE_CONF_OPTS += --disable-bsdcat
+endif
+
 ifeq ($(BR2_PACKAGE_ACL),y)
 LIBARCHIVE_DEPENDENCIES += acl
 else
@@ -99,4 +109,23 @@ else
 LIBARCHIVE_CONF_OPTS += --without-lzma
 endif
 
+# The only user of host-libarchive needs zlib support
+HOST_LIBARCHIVE_DEPENDENCIES = host-zlib
+HOST_LIBARCHIVE_CONF_OPTS = \
+	--disable-bsdtar \
+	--disable-bsdcpio \
+	--disable-bsdcat \
+	--disable-acl \
+	--disable-xattr \
+	--without-bz2lib \
+	--without-expat \
+	--without-libiconv-prefix \
+	--without-xml2 \
+	--without-lzo2 \
+	--without-nettle \
+	--without-openssl \
+	--without-lz4 \
+	--without-lzma
+
 $(eval $(autotools-package))
+$(eval $(host-autotools-package))
-- 
2.7.4

